File: iosApp/Views/Onboarding/Components/ConsentCheckbox.swift
import SwiftUI

/// A checkbox bound to a Bool. Used on the Hippocratic License and
/// medical-device disclaimer steps where the spec requires an explicit
/// "I have read and accept" acknowledgement before Continue activates.
struct ConsentCheckbox: View {

    @Binding var isChecked: Bool
    let label: String

    var body: some View {
        Button {
            isChecked.toggle()
        } label: {
            HStack(alignment: .top, spacing: 12) {
                Image(systemName: isChecked ? "checkmark.square.fill" : "square")
                    .foregroundStyle(isChecked ? Color.accentColor : Color.secondary)
                    .font(.title3)
                Text(label)
                    .multilineTextAlignment(.leading)
                    .foregroundStyle(.primary)
                Spacer(minLength: 0)
            }
        }
        .buttonStyle(.plain)
        .accessibilityElement(children: .combine)
        .accessibilityAddTraits(isChecked ? [.isSelected] : [])
    }
}

File: androidApp/src/main/kotlin/com/malaria/android/ui/onboarding/components/ConsentCheckbox.kt
package com.malaria.android.ui.onboarding.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Checkbox
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

/**
 * A checkbox bound to a Boolean. Used on the Hippocratic License and
 * medical-device disclaimer steps where the spec (§10) requires an explicit
 * "I have read and accept" acknowledgement before Continue activates.
 *
 * Mirrors `iosApp/Views/Onboarding/Components/ConsentCheckbox.swift`.
 */
@Composable
fun ConsentCheckbox(
    checked: Boolean,
    onCheckedChange: (Boolean) -> Unit,
    label: String,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clickable { onCheckedChange(!checked) }
            .padding(vertical = 8.dp),
        verticalAlignment = Alignment.Top,
        horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Checkbox(checked = checked, onCheckedChange = onCheckedChange)
        Text(
            text = label,
            style = MaterialTheme.typography.bodyMedium,
            modifier = Modifier.padding(top = 12.dp),
        )
    }
}
